### ISSUE TITLE
Warn or error when not enough colors are set for a categorically shaded plot

### ISSUE THREAD

**Issue by maximlt** [holoviz/holoviews#5374]
In https://github.com/holoviz/hvplot/pull/759 it was noted that:

> Note that datashader can happily render hundreds of thousands of columns, but it will run out of colors for them after 22 columns by default. Even with Glasbey categorical maps from Colorcet, it will run out after 256, raising a fatal error. It might be good to catch that error in hvPlot and either drop back to non-categorical with a warning, or at least raise a useful error telling people what to do:
> ```Too many columns for the number of colors in the `color_key` ({len(color_key)}; 
reverting to a non-categorical plot. This warning can be suppressed by specifying 
`categorical=False` (??) or by supplying a `color_key` with at least {len(columns)} 
colors.```

It was suggested that instead of adding this warning/error at hvPlot's level, it could be done at HoloViews' level with a message that would be informative enough to hvPlot's users. The message returned by Datashader is indeed not that clear: `ValueError: Insufficient colors provided (256) for the categorical fields available (299)`

**Comment by jlstevens** [comment]
> ... it could be done at HoloViews' level with a message that would be informative enough to hvPlot's users.

The goal would be to write the message so that it makes sense both to hvplot users and people using HoloViews directly.

**Comment by jbednar** [comment]
Right. My proposed message also doesn't explain that people who have tons of traces and still want categorical colors should create a category column, categorize each trace, and then try again; one color per trace won't work well in such cases. So maybe:

```
Too many columns for the number of colors in the `color_key` ({len(color_key)}; 
reverting to a non-categorical plot. This warning can be suppressed by specifying 
`categorical=False` (??), by supplying a `color_key` with at least {len(columns)} colors,
or by grouping by a categorical column that matches the number of colors available.
```

That's pretty wordy and maybe not all that clear. Plus for hvPlot it would mention that categorical grouping can be done with the `by` keyword; not sure how to do that in HoloViews.

**Comment by jlstevens** [comment]
> That's pretty wordy and maybe not all that clear.

Agreed but it is still better than the old error! Other than maybe a reference to the docs (which can be tricky!) I can't think of how to improve this right now.

There is a discussion to be had about whether HoloViews could reference online docs more in docstrings (like panel does).

**Comment by poleguy** [comment]
I just ran into this error message using dsshow with matplotlib.
I don't know how to find the answer easily via the documentation, google, etc. This github was my only hope, but it leaves me with questions. At least it points me to the need for adding a color_key parameter. I'm in support of a better error message.

**Comment by poleguy** [comment]
Note to future me. I fixed this by adding color_key to dsshow. Something like:
```
import colorcet as cc
colors = cc.glasbey_cool
artist4 = dsshow(
    df, 
    ds.glyphs.LineAxis0('x', 'y'), 
    ds.count_cat('cat'),
    color_key=colors,
    ax=ax
)
```
